fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -1473,7 +1473,9 @@ private void beginTxn(String dbName, String tblName) throws UserException, TExce
             InterruptedException, ExecutionException, TimeoutException {
         TransactionEntry txnEntry = context.getTxnEntry();
         TTxnParams txnConf = txnEntry.getTxnConf();
-        long timeoutSecond = ConnectContext.get().getSessionVariable().getQueryTimeoutS();
+        ConnectContext ctx = ConnectContext.get();
+        int timeoutSecond = Math.max(ctx.getSessionVariable().getInsertTimeoutS(), ctx.getExecTimeout());
+        ctx.setExecTimeout(timeoutSecond);
```

Comment:
And just use 
```java
long timeoutSecond = ConnectContext.get().getExecTimeout
```
to fetch the required exec_tiemout

Reply:
yes, here no need to set